### ISSUE TITLE
AzureManagedMachinePool CRD spec not being respected in cluster resource.

### ISSUE THREAD

**Issue by TheNakedZealot** [kubernetes-sigs/cluster-api-provider-azure#2059]
/kind bug

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
I applied a yaml including a AzureManagedMachinePool manifest, such as:

```
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: AzureManagedMachinePool
metadata:
  name: agentpool1
  namespace: default
spec:
  maxPods: 250
  mode: User
  osDiskSizeGB: 512
  sku: Standard_F64s_v2
  scaling:
    minSize: 2
    maxSize: 24
```
The generated CRD reflected this manifest...
```
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: AzureManagedMachinePool
metadata:
  creationTimestamp: "2022-02-02T18:54:57Z"
  finalizers:
  - azurecluster.infrastructure.cluster.x-k8s.io
  generation: 1
  labels:
    azuremanagedmachinepool.infrastructure.cluster.x-k8s.io/agentpoolmode: User
    cluster.x-k8s.io/cluster-name: arch-developer-enablement
    kustomize.toolkit.fluxcd.io/name: arch-developer-enablement
    kustomize.toolkit.fluxcd.io/namespace: default
  name: agentpool1
  namespace: default
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: MachinePool
    name: agentpool1
    uid: 0c438532-e5c5-4aea-8c28-5969894b462b
  resourceVersion: "5121703"
  uid: 4021039b-8a4f-4e15-b6c1-0eee6b07e94b
spec:
  maxPods: 250
  mode: User
  name: agentpool1
  osDiskSizeGB: 512
  scaling:
    maxSize: 24
    minSize: 2
  sku: Standard_F64s_v2
```
But the actual Cluster resource only respected some fields in the spec.
```
Addresses:
  Hostname:    aks-agentpool1-35511615-vmss000000
Capacity:
  cpu:                64
  ephemeral-storage:  520139172Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             132007916Ki
  pods:               30
Allocatable:
  cpu:                63260m
  ephemeral-storage:  479360260122
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             121467884Ki
  pods:               30
```

**What did you expect to happen:**

I expected the cluster resource with the `maxPods` value of 250, and with AKS autoscaling enabled on the cluster with a minimum of 2 'agentpool1' worker nodes. The created resource does not reflect that.

![image](https://user-images.githubusercontent.com/53411346/153086737-6ad30e3d-246e-4b2c-aa10-8724da11dfc9.png)

**Anything else you would like to add:**

Confimed options exist here: https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api-provider-azure/infrastructure.cluster.x-k8s.io/AzureManagedMachinePool/v1beta1@v1.1.0

**Environment:**

- cluster-api-provider-azure version: 1.1.1
- Kubernetes version: (use `kubectl version`): 1.22.4
- OS (e.g. from `/etc/os-release`): Ubuntu

**Comment by jackfrancis** [comment]
@zmalik can you validate that you fixed this in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1945 ?

**Comment by CecileRobertMichon** [comment]
@TheNakedZealot this should be fixed in release v1.1.2. Feel free to open a new issue if you run into more trouble.

/close
